macantine/tasks.py

```diff
@@ -258,3 +260,110 @@ def delete_old_historical_records():
         return
     logger.info(f"History items older than {settings.MAX_DAYS_HISTORICAL_RECORDS} days will be removed.")
     call_command("clean_old_history", days=settings.MAX_DAYS_HISTORICAL_RECORDS, auto=True)
+
+
+def _flatten_declared_data(df):
+    tmp_df = pd.json_normalize(df["declared_data"])
+    df = pd.concat([df.drop("declared_data", axis=1), tmp_df], axis=1)
+    return df
+
+
+def _extract_dataset_teledeclaration(year):
+    td_columns = [
+        "id",
+        "applicant_id",
+        "teledeclaration_mode",
+        "creation_date",
+        "year",
+        "version",
+        "canteen.id",
+        "canteen.siret",
+        "canteen.name",
+        "canteen.central_producer_siret",
+        "canteen.department",
+        "canteen.region",
+        "canteen.satellite_canteens_count",
+        "canteen.economic_model",
+        "canteen.management_type",
+        "canteen.production_type",
+        "canteen.sectors",
+        "canteen.line_ministry",
+        "teledeclaration_ratio_bio",
+        "teledeclaration_ratio_egalim_hors_bio",
+    ]
+    td = pd.DataFrame(Teledeclaration.objects.filter(year=year).values())
+    td = _flatten_declared_data(td)
+    td["teledeclaration_ratio_bio"] = td["teledeclaration.value_bio_ht"] / td["teledeclaration.value_total_ht"]
+    td["teledeclaration_ratio_egalim_hors_bio"] = (
+        td["teledeclaration.value_sustainable_ht"] / td["teledeclaration.value_total_ht"]
+    )
+    td = td.loc[:, ~td.columns.duplicated()]
+    td = td[td_columns]
+    td.columns = td.columns.str.replace(".", "_")
+    td = td.reset_index(drop=True)
+    return td
+
+
+def _extract_dataset_canteen():
+    canteens_col = [
+        "id",
+        "name",
+        "siret",
+        "city",
+        "city_insee_code",
+        "postal_code",
+        "department",
+        "region",
+        "economic_model",
+        "management_type",
+        "production_type",
+        "satellite_canteens_count",
+        "central_producer_siret",
+        "creation_date",
+        "daily_meal_count",
+        "yearly_meal_count",
+        "line_ministry",
+        "modification_date",
+        "publication_status",
+        "logo",
+        "sectors",
+    ]
+    all_canteens = Canteen.objects.all()
+    active_canteens_id = [c.id for c in all_canteens if not c.can_be_claimed]
```

Comment:
Comment ça se passe niveau perf avec les 26k cantines qu'on a ? Disons, un `for` comme celui ci vs un truc directement en base comme `Canteens.objects.exclude(managers=None)` :thinking:

Reply:
C'est beaucoup mieux ta méthode ! Je n'y ai pas pensé